### PR TITLE
test(e2e): add support to build images which generate coverage data

### PR DESCRIPTION
**Pull request by blaisedias**

### CODE REVIEW

**Review by chriswldenyer — APPROVED** [review]
LGTM

**Review by gila — CHANGES_REQUESTED** [review]
These coverage aspects do not belong in the YAML files and should be part of the testing bits. 

Also, why do we need a new image type and don't we set the RUSTFLAGS as an argument?

File: scripts/release.sh [hunk 1]
```diff
@@ -108,20 +110,32 @@ while [ "$#" -gt 0 ]; do
...
+      shift
+      ;;
     *)
       echo "Unknown option: $1"
       exit 1
       ;;
   esac
 done
 
+if [ -n "$COVERAGE" ] && [ -n "$DEBUG" ]; then
```
Comment (gila) [inline scripts/release.sh:124]: Why not ?

**Review by gila — CHANGES_REQUESTED** [review]

File: scripts/release.sh (hunk 1 above)
Comment (blaisedias) [reply scripts/release.sh:124]: see https://github.com/openebs/mayastor/pull/1007 which you approved

File: nix/pkgs/mayastor/cargo-package.nix [hunk 2]
```diff
@@ -83,15 +87,25 @@ let
...
-    buildType = "release";
-    buildInputs = buildProps.buildInputs ++ [ libspdk ];
-    SPDK_PATH = "${libspdk}";
-  });
-  debug = rustPlatform.buildRustPackage (buildProps // {
-    buildType = "debug";
-    buildInputs = buildProps.buildInputs ++ [ libspdk-dev ];
-    SPDK_PATH = "${libspdk-dev}";
-  });
+  release = rustPlatform.buildRustPackage
```
Comment (gila) [inline nix/pkgs/mayastor/cargo-package.nix:90]: Why is the indentation changed here?

File: scripts/release.sh [hunk 3]
```diff
@@ -30,6 +30,7 @@ Options:
   -d, --dry-run              Output actions that would be taken, but don't run them.
   -h, --help                 Display this text.
   --registry <host[:port]>   Push the built images to the provided registry.
+  --coverage                 Build coverage-enabled version of images where possible.
```
Comment (gila) [inline scripts/release.sh:33]: Shall we just add the coverage option to debug builds?

File: csi/src/server.rs [hunk 4]
```diff
@@ -237,12 +237,29 @@ async fn main() -> Result<(), String> {
...
+    let mut signal_term = tokio::signal::unix::signal(
+        tokio::signal::unix::SignalKind::terminate(),
+    )
+    .expect("Failed to egister handler for SIGTERM");
+    let mut signal_int = tokio::signal::unix::signal(
+        tokio::signal::unix::SignalKind::interrupt(),
+    )
+    .expect("Failed to register handler for SIGINT");
+
+    tokio::select! {
```
Comment (gila) [inline csi/src/server.rs:249]: In line with comments elsewhere, I do not understand or see, why it helps to override the signal handler with nothing.

**Review by gila — CHANGES_REQUESTED** [review]

File: scripts/release.sh (hunk 3 above)
Comment (blaisedias) [reply scripts/release.sh:33]: The first requirement is for coverage builds matching release code.
That is what this change is for.

File: nix/pkgs/mayastor/cargo-package.nix (hunk 2 above)
Comment (blaisedias) [reply nix/pkgs/mayastor/cargo-package.nix]: fixed https://github.com/openebs/mayastor/pull/1041/commits/778fc7c2c74269b75e51c5e6f809f8a4466576e6

**Pull request closed without merging**